### PR TITLE
Align StatisticsImpl / StatisticsData

### PR DESCRIPTION
**Pull request by grooverdan**
Pinned the alignment of StatisticsData to the cacheline size rather than just extending its size (which could go over two cache lines)if unaligned in allocation.

Avoid compile errors in the process as per individual commit messages.

strengthen static_assert to CACHELINE rather than the highest common multiple.

### CODE REVIEW

**Comment by grooverdan** [comment]
faligned-new is the gcc/clang fix. The operator overrides are actually only needed on Visual Studio.

**Comment by yiwu-arbug** [comment]
@ajkr do you have concern if I land this?

**Comment by yiwu-arbug** [comment]
@grooverdan would be great if you can include any benchmarking result with the PR. Thanks.

**Comment by grooverdan** [comment]
@yiwu-arbug , @ajkr  thanks for the review and merge. I will get to benchmarking at some point.

File: Makefile [hunk 1]
```diff
@@ -98,6 +98,10 @@ OPT += -momit-leaf-frame-pointer
 endif
 endif
 
+ifeq (,$(shell $(CXX) -fsyntax-only -faligned-new -xc++ /dev/null 2>&1))
+CXXFLAGS += -faligned-new -DHAVE_ALIGNED_NEW
```
Comment (yiwu-arbug) [inline Makefile]: @grooverdan  We start to get the following compile error with clang_analyzer (clang-6.0):
```
cache/lru_cache.cc:118: error: undefined reference to 'operator delete(void*, std::align_val_t)'
monitoring/statistics.cc:28: error: undefined reference to 'operator delete(void*, std::align_val_t)'
```
The destructors are now calling c++17 delete operator, and seems to related to the faligned-new flag. Mind take a look? Thanks.

Comment (grooverdan) [reply Makefile]: sure. Added #4075 to cover compile testing of clang-6,7,gcc-8

Comment (grooverdan) [reply Makefile]: what was the compile that triggered it?

Comment (miasantreble) [reply Makefile]: you can try 
```
USE_CLANG=1 OPT=-g make -j1 analyze 
```

**Pull request closed without merging**